src/library_wasm_worker.js

```diff
@@ -292,18 +292,24 @@ Atomics.waitAsync = (i32a, index, value, maxWaitMilliseconds) => {
   $liveAtomicWaitAsyncs: '{}',
   $liveAtomicWaitAsyncCounter: '0',
 
-  emscripten_atomic_wait_async__deps: ['$atomicWaitStates', '$liveAtomicWaitAsyncs', '$liveAtomicWaitAsyncCounter', '$jstoi_q'],
+  emscripten_atomic_wait_async__deps: ['$atomicWaitStates', '$liveAtomicWaitAsyncs', '$liveAtomicWaitAsyncCounter', '$jstoi_q',
+#if !MINIMAL_RUNTIME
+  '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+#endif
```

Comment:
I believe this is not needed. The compiler adds the runtime-keepalive stuff automatically.

Reply:
Unfortunately, after testing, the compiler does not automatically add runtime-keepalive stuff, because `library_wasm_worker.js` is not in the core system libraries of `modules.js`, so need to add a reference here manually.
https://github.com/emscripten-core/emscripten/blob/f9ce05b52f82e3c644eafd5d4a6d26e8898518af/src/modules.js#L37

Reply:
Hmm, it's not in the core system libraries always linked against, but when you enable wasm workers then it is added to the list. And then it should be processed like any other library. That is, there should be no difference between a core library and a non-core library if both are linked in. Is that not so?

Reply:
> Hmm, it's not in the core system libraries always linked against, but when you enable wasm workers then it is added to the list. And then it should be processed like any other library. That is, there should be no difference between a core library and a non-core library if both are linked in. Is that not so?

According to the `modules.js`, when enable WASM worker support, it doesn't seem to be automatically added to the core library.
As a result, dependencies must be added manually.

Reply:
Where do you see that? (I see no mention of wasm workers in `modules.js`)

Reply:
I think @kripken is right, we hopefully don't need these `__deps` here..  if you remove the `__deps` here what error do you see?